azkaban-web-server/src/main/java/azkaban/webapp/servlet/ProjectManagerServlet.java

```diff
@@ -1749,7 +1754,7 @@ private void ajaxHandleUpload(HttpServletRequest req,
           error =
               error.substring(0, 512) + "<br>Too many errors to display.<br>";
         }
-        ret.put("error", "Installation Failed.<br>" + error);
```

Comment:
It is impossible from here to tell if it is 400 or 500. I suppose we should err on the side of caution and return 500.